packages/expo-router/src/modal/ModalContext.tsx

```diff
@@ -0,0 +1,224 @@
+        }
+      });
+      prevModalConfigs.current = modalConfigs;
+    }
+  }, [modalConfigs]);
+
+  const openModal = useCallback(
+    (config: ModalConfig) => setModalConfigs((prev) => [...prev, config]),
+    []
+  );
+
+  const emitCloseEvent = useCallback((id: string) => {
+    closeEventListeners.current.forEach((callback) => callback(id));
+  }, []);
+
+  const emitShowEvent = useCallback((id: string) => {
+    showEventListeners.current.forEach((callback) => callback(id));
+  }, []);
+
+  const closeModal = useCallback((id: string) => {
+    setModalConfigs((prev) => {
+      const modalIndex = prev.findIndex((config) => config.uniqueId === id);
+      if (modalIndex >= 0) {
+        return prev.filter((_, index) => index < modalIndex);
+      }
+      return prev;
+    });
+  }, []);
+
+  const addEventListener = useCallback((type: 'close' | 'show', callback: (id: string) => void) => {
+    if (type !== 'close' && type !== 'show') return () => {};
+
+    if (!callback) {
+      console.warn('Passing undefined as a callback to addEventListener is forbidden');
+      return () => {};
+    }
+
+    const eventListeners = type === 'close' ? closeEventListeners : showEventListeners;
+    eventListeners.current.add(callback);
+
+    return () => {
+      eventListeners.current.delete(callback);
+    };
+  }, []);
+
+  const rootId = useMemo(() => nanoid(), []);
+
+  return (
+    <ScreenStack style={styles.stackContainer}>
+      <ScreenStackItem
+        screenId={rootId}
+        activityState={2}
+        style={StyleSheet.absoluteFill}
+        headerConfig={{
+          hidden: true,
+        }}>
+        <ModalContext.Provider
+          value={{
+            modalConfigs,
+            openModal,
+            closeModal,
+            addEventListener,
+          }}>
+          {children}
+        </ModalContext.Provider>
+      </ScreenStackItem>
+      {modalConfigs.map((config) => (
+        <ScreenStackItem
+          key={config.uniqueId}
+          {...config.viewProps}
+          screenId={`${rootId}${config.uniqueId}`}
+          activityState={2}
+          stackPresentation={getStackPresentationType(config)}
+          stackAnimation={getStackAnimationType(config)}
+          nativeBackButtonDismissalEnabled
+          headerConfig={{
+            hidden: true,
+          }}
+          contentStyle={{
+            flex: 1,
+            backgroundColor: config.transparent ? 'transparent' : 'white',
+          }}
+          sheetAllowedDetents={config.detents}
+          style={[
+            StyleSheet.absoluteFill,
+            {
+              backgroundColor: config.transparent ? 'transparent' : 'white',
+            },
+          ]}
+          onDismissed={() => {
+            // On Android, when formSheet is dismissed, onDisappear is not called.
+            if (Platform.OS === 'android' && config.presentationStyle === 'formSheet') {
```

Comment:
```suggestion
            if (process.env.EXPO_OS === 'android' && config.presentationStyle === 'formSheet') {
```
Should avoid Platform.OS as it's an expensive and unreliable babel transform that adds an extra import to an expensive package.